lib/logstash/inputs/couchdb_changes.rb

```diff
@@ -181,7 +190,8 @@ def build_uri
 
   private
   def reconnect?
-    sleep(@always_reconnect ? @reconnect_delay : 0)
+    # Loop each second so stop? can work its magic
+    @reconnect_delay.times { sleep(1) unless stop? } if @always_reconnect
```

Comment:
You'll need to have the gemspec depend on the latest version of `stud` or greater. Which I believe is `>= 0.0.22`, and that should be it!